ui/lib/components/AutoRefreshButton/AutoRefreshButton.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { DownOutlined, LoadingOutlined, SyncOutlined } from '@ant-design/icons'
+import { Spin, Dropdown, Menu, Space } from 'antd'
+import { useSpring, animated } from 'react-spring'
+import { getValueFormat } from '@baurine/grafana-value-formats'
+import { useTranslation } from 'react-i18next'
+import { addTranslationResource } from '@lib/utils/i18n'
+import { useGetSet } from 'react-use'
+
+interface AutoRefreshButtonProps {
+  // use seconds as options
+  options: number[]
+  autoRefreshSeconds: number
+  onAutoRefreshSecondsChange: (v: number) => void
+  onRefresh: () => void
+  isLoading: boolean
+  enabled?: boolean
+}
+
+const translations = {
+  en: {
+    refresh: 'Refresh',
+    auto_refresh: {
+      title: 'Auto Refresh',
+      off: 'Off',
+    },
+  },
+  zh: {
+    refresh: '刷新',
+    auto_refresh: {
+      title: '自动刷新',
+      off: '关闭',
+    },
+  },
+}
+
+for (const key in translations) {
+  addTranslationResource(key, {
+    component: {
+      autoRefreshButton: translations[key],
+    },
+  })
+}
+
+export function AutoRefreshButton({
+  options,
+  autoRefreshSeconds,
+  onAutoRefreshSecondsChange,
+  onRefresh,
+  enabled = true,
+  isLoading,
+}: AutoRefreshButtonProps) {
+  const { t } = useTranslation()
+  const autoRefreshMenu = (
+    <Menu
+      onClick={({ key }) => onAutoRefreshSecondsChange(parseInt(key as string))}
+      selectedKeys={[String(autoRefreshSeconds || 0)]}
+    >
+      <Menu.ItemGroup
+        title={t('component.autoRefreshButton.auto_refresh.title')}
+      >
+        <Menu.Item key="0">
+          {t('component.autoRefreshButton.auto_refresh.off')}
+        </Menu.Item>
+        <Menu.Divider />
+        {options.map((sec) => {
+          return (
+            <Menu.Item key={String(sec)}>
+              {getValueFormat('s')(sec, 0)}
+            </Menu.Item>
+          )
+        })}
+      </Menu.ItemGroup>
+    </Menu>
+  )
+
+  const handleRefresh = useCallback(() => {
+    if (isLoading) {
+      return
+    }
+    onRefresh()
+  }, [isLoading, onRefresh])
+
+  // Auto refresh
+  const [timer, setTimer] = useState<number | undefined>()
```

Comment:
There is a specific type for the return value of `setTimeout` and maybe you can use it.